src/app/Http/Actions/AccountRegistration/ActivateUnverifiedAccount.php

```
@@ -0,0 +1,52 @@
+<?php
+
+namespace App\Http\Actions\AccountRegistration;
+
+use App\Entities\Accounts\Repositories\UnactivatedAccountRepository;
+use App\Entities\Accounts\Repositories\AccountRepository;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Carbon;
+
+final class ActivateUnverifiedAccount
+{
+    private $unactivatedAccountRepository;
+    private $accountRepository;
+    
+    public function __construct(UnactivatedAccountRepository $unactivatedAccountRepository, AccountRepository $accountRepository) {
+        $this->unactivatedAccountRepository = $unactivatedAccountRepository;
+        $this->accountRepository = $accountRepository;
+    }
+
+    public function execute(string $email, string $ip)
+    {
+        $unactivatedAccount = $this->unactivatedAccountRepository->getByEmail($email);
+        if ($unactivatedAccount === null) {
+            throw new \Exception('This unactivated account does not exist');
+        }
+
+        $accountByEmail = $this->accountRepository->getByEmail($email);
+        if ($accountByEmail) {
+            // Account already activated
+            abort(410);
+        }
+
+        DB::beginTransaction();
```

Is there a reason for this versus the [closure method](https://laravel.com/docs/5.8/database#database-transactions), which essentially does the same thing?

---

Using the closure will consume the exception and fail silently. If an error we didn't expect happens, it's better that it fails and we get notified via Sentry

---

Ah right, I would have thought there'd have been some way of getting Sentry to still report it but some quick googling didn't find anything